apis/dataprotection/v1alpha1/actionset_types.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ActionSetSpec defines the desired state of ActionSet
+type ActionSetSpec struct {
+	// backupType specifies the backup type, supported values: Full, Continuous.
+	// Full means full backup.
+	// Incremental means back up data that have changed since the last backup (full or incremental).
+	// Differential means back up data that have changed since the last full backup.
+	// Continuous will back up the transaction log continuously, the PITR (Point in Time Recovery).
+	// can be performed based on the continuous backup and full backup.
+	// +kubebuilder:validation:Enum={Full,Incremental,Differential,Continuous}
+	// +kubebuilder:default=Full
+	// +kubebuilder:validation:Required
+	BackupType BackupType `json:"backupType"`
```

Comment:
Since you have refactored to a generic name (ActionSet), and I would suggest have this attribute as "type" to accommodate future extensible design? Or otherwise this API would be named as "BackupRestoreActionSet".